components/program/Timeline.vue

```diff
@@ -0,0 +1,355 @@
+<script setup lang="ts">
+  import type { AlertType } from 'composables/useAlerts'
+
+  const useAuth = useAuthStore()
+
+  const companies = await $fetch('/api/company', { method: 'GET' })
+
+  const events = ref({})
+  const eventsByDate: Ref<{ [key: string]: any[] }> = ref({})
+  const dates = ref([])
+  const fetchAndUpdateEvents = async () => {
+    try {
+      const updatedEvents = await getEvents()
+      events.value = updatedEvents
+      eventsByDate.value = groupEventsByDateStart(updatedEvents)
+      dates.value = Object.keys(eventsByDate.value)
+      state.selectedDate = dates.value[0] || ''
+    } catch (err) {
+      console.error(err)
```

Comment:
Should use an alert and redirect to the home page instead.

Reply:
Users might, want to sign up for multiple events, so I don't think it should navigate to the main page again